spec/ca_testing/extensions/selenium/webdriver/logs_spec.rb

```diff
@@ -0,0 +1,117 @@
+# frozen_string_literal: true
+
+describe Selenium::WebDriver::Logs do
+  before do
+    if Gem::Version.new(Selenium::WebDriver::VERSION) < Gem::Version.new("4.0.0.beta1")
+      skip("Only supported across all driver types from Webdriver beta2")
+    end
+  end
+
+  # The Bridge is considered API Private, but we need to mock it!
+  let(:bridge) { session.driver.browser.logs.instance_variable_get(:@bridge) }
+  let(:log_entry) { instance_double(Selenium::WebDriver::LogEntry, to_s: "Time - SEV - Browser or Driver Message") }
+  let(:session) { Capybara::Session.new(:selenium_chrome_headless) }
+  let(:logs) { session.driver.browser.logs }
+  before do
+    allow(bridge)
+      .to receive(:log)
+      .with(type)
+      .and_return([log_entry])
+
+    allow(logs)
+      .to receive(:cached_logs)
+      .and_call_original
+  end
+
+  context "for browser logging" do
+    let(:type) { :browser }
+
+    describe "#get" do
+      it "caches the response from browser logs" do
```

Comment:
I'm not sure this test covers everything we could be testing. Rather than testing that `cached_logs` is called could you stub the @bridge and test whats returned from there?

Reply:
Sadly not, as that would then be testing the execution of the api payload request. I cannot mimic fetching logs in rspec so instead I just stub the return value from the logs.

Here the two new things I've added are that we can write to a file and cache the logs (Which is needed, because the fetch is destructive for reasons unbeknownst to me)

Also it's worth pointing out that the existing code `@bridge.log type` is literally a copy and paste (super if you will).